src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] DEFAULT_ARRAYLIST = {};
+    private Object[] objectArray;
+
+    private int size;
+
+    public ArrayList() {
+        objectArray = DEFAULT_ARRAYLIST;
+    }
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            this.objectArray = new Object[initialCapacity];
+        } else if (initialCapacity == 0) {
+            this.objectArray = DEFAULT_ARRAYLIST;
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity: "
+                    + initialCapacity);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        if (size < index || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Cen`t add element. "
+                    + "Index " + index + " is out of range 0 - " + size);
+        }
+        if (objectArray.length == 0) {
+            objectArray = new Object[DEFAULT_CAPACITY];
+        }
 
+        if (objectArray.length < size + 1) {
+            objectArray = Arrays.copyOf(objectArray, objectArray.length
+                    + (objectArray.length >> 1));
```

Comment:
Let's create a separate private method for this part. It would be easier to read your code if this part has name like: grow or increaseArray